Implemented the movement tutorial and cutscene code

Megafont opened the pull request:
HOW TO TEST:
1. Go to the Drone game object
2. Open the MovementTutorial child object
3. Go to the Tutorial_Movement01_PlayCutScene component in the inspector
4. Disable the SkipTutorial checkbox
5. Disable the SkipCutScene checkbox to test the CutScenePlayer


NOTES:
+ Movement tutorial part 3 script has a bool setting that lets you set whether the player must hit all four movement keys (WASD) to satisfy the mobility test
+ The scripts for the four parts of the movement tutorial have some settings that can tweak delays, like how long the drone waits before it flies onto the screen
+ Added a LogTransitions public bool property to the StateMachineInstance class so I could turn that option on easier.

Megafont committed:
Implemented the movement tutorial

+ Movement tutorial part 3 script has a bool setting that lets you set whether the player must hit all four movement keys to satisfy the mobility test
+ The scripts for the four parts of the movement tutorial have some settings that can tweak delays, like how long the drone waits before it flies onto the screen
+ Added a LogTransitions public bool property to the StateMachineInstance class so I could turn that option on easier.

Megafont committed:
Forgot to enable SkipTutorial, so this commit fixes that

CaseyWalshWarder commented:
This has a merge conflict after merging in https://github.com/P1Gaming/SkyCore/pull/35

CaseyWalshWarder commented on Assets/Scripts/Drone/DroneMovement.cs:
```diff
@@ -30,11 +45,22 @@ public void IdleMovement()
     {
         float maxMovementDistance = Time.deltaTime * _movementSpeed;
+
+        float distance = Vector3.Distance(transform.position, transform.position + toTarget);
```
You can just use toTarget.magnitude rather than Vector3.Distance

CaseyWalshWarder commented on Assets/Scripts/Drone/DroneMovement.cs:
```diff
@@ -61,6 +87,40 @@ public Vector3 FromDroneToNear(Transform toHoverNear, bool backAwayIfTooClose =
+
+        //Debug.DrawLine(transform.position, transform.position + direction, Color.green, 1f);
+
+        if (direction.magnitude > -_hasArrivedDistanceThreshold &&
```
The first part of the condition will always be true, because a vector's magnitude is never negative.

CaseyWalshWarder commented on Assets/Scripts/Drone/Tutorial/Tutorial_Movement02_DroneInterceptPlayer.cs:
```diff
@@ -0,0 +1,145 @@
+
+        if (_medicalAssistanceComplete)
+        {
+            // Trigger transition to next part of the tutorial.
```
Maybe you could just use a trigger parameter to advance to the next step. That way it's easier to make changes like inserting another step.

CaseyWalshWarder commented on Assets/Scripts/Drone/Tutorial/Tutorial_Movement01_PlayCutScene.cs:
```diff
@@ -0,0 +1,114 @@
+    private float _droneDistanceToStopInFrontOfPlayer;
+    [Tooltip("This specifies how long of a delay between the end of the cutscene and when the drone flies onto the screen to help the player.")]
+    [SerializeField, Min(0)]
+    private float _droneInterceptDelay = 3f;
```
It might be possible to use min duration on transitions to remove some of the time tracking code.

Megafont replied:
I couldn't do that here, as the outer if statement means it wouldn't work.

CaseyWalshWarder commented on Assets/Scripts/Drone/Tutorial/Tutorial_Movement04_MobilityTest.cs:
```diff
@@ -0,0 +1,214 @@
+        }
+        
+
+        return false;
```
I think this method can return void.

CaseyWalshWarder reviewed (commented):
I haven't checked for bugs yet because of the merge conflict, but the code looks good. I left some suggestions but there are no major issues which need to be resolved to merge it in.

Megafont committed:
Implemented suggestions and cutscene code

I didn't end up using MinDuration as suggested, as the outer if statement there means it wouldn't work. All other suggestions are implemented, though.

I added a new SkipCutscene field to Tutorial_Movement_PlayCutScene.
If disabled, it will play a test cutscene for now, which can also be played by setting the CutSceneToPlayOnStart field of the CutScenePlayer object in the scene. Just drag the TestCutScene object into that field (its a child object of the CutScenePlayer).

CaseyWalshWarder commented:
Looks good. Just don't forget to fix the merge conflict. Generally with scenes I just update from develop, use the new file, and re-apply the changes I made.
